tests/framework-tests/preferences/js/SeparatedPanelPrefsEditorResponsiveTests.js

```diff
@@ -0,0 +1,218 @@
+        var attrState = state ? "true" : "false";
+
+        fluid.tests.assertAriaForToggleButton(toggleButton, "Hide/show", panelId, state);
+        jqUnit.assertEquals("Panel has the group role", "group", panel.attr("role"));
+        jqUnit.assertEquals("Panel has the correct aria-label", that.options.strings.panelLabel, panel.attr("aria-label"));
+        jqUnit.assertEquals("Panel has correct aria-expanded", attrState, panel.attr("aria-expanded"));
+    };
+
+    fluid.tests.assertResetButton = function (separatedPanel, state) {
+        separatedPanel.locate("reset").each(function (idx, elm) {
+            elm = $(elm);
+            var smallScreenContainer = elm.parents(".fl-panelBar-smallScreen");
+
+            if (smallScreenContainer.length > 0) {
+                jqUnit.assertEquals("The small screen Reset button visibility should be " + state, state, elm.is(":visible"));
+                fluid.tests.assertAriaForButton(elm, "The small screen Reset", separatedPanel.slidingPanel.panelId);
+            } else {
+                jqUnit.assertEquals("The wide screen Reset button visibility should be false", false, elm.is(":visible"));
+                fluid.tests.assertAriaForButton(elm, "The wide screen Reset", separatedPanel.slidingPanel.panelId);
+            }
+        });
+    };
+
+    fluid.tests.assertSeparatedPanelState = function (separatedPanel, state) {
+        jqUnit.assertEquals("The iframe visibilith should be" + state, state, separatedPanel.iframeRenderer.iframe.is(":visible"));
+        fluid.tests.assertResetButton(separatedPanel, state);
+        fluid.tests.assertAria(separatedPanel.slidingPanel, state);
+    };
+
+    fluid.tests.assertSeparatedPanelInit = function (separatedPanel) {
+        fluid.tests.assertSeparatedPanelState(separatedPanel, false);
+        fluid.tests.prefs.assertPresent(separatedPanel, fluid.tests.prefs.expectedSeparatedPanel);
+        fluid.tests.prefs.assertPresent(separatedPanel.prefsEditor, fluid.tests.prefs.expectedComponents["fluid.prefs.separatedPanel"]);
+    };
+
+    fluid.tests.assertPanelVisibility = function (prefsEditor, testName, panelIndex) {
+        var panels = prefsEditor.locate("panels");
+
+        panels.each(function (idx, elm) {
+            var panelOffset = $(elm).offset().left;
+            if (idx === panelIndex) {
+                jqUnit.assertEquals(testName + ": The panel at index " + idx + " should be scrolled into view with offset = 0", 0, panelOffset);
+            } else {
+                jqUnit.assertNotEquals(testName + ": The panel at index " + idx + " should not be scrolled into view and have an offset greater or less than 0", 0, panelOffset);
+            }
+        });
+    };
+
+    fluid.tests.direction = {
+        LEFT: false,
+        RIGHT: true
+    };
+
+    fluid.tests.clickArrow = function (elm, direction) {
+        var keyEvent = $.Event("click");
+        keyEvent.offsetX = direction ? elm.width() : 0;
```

Comment:
It's a bit indirect and confusing to use `{LEFT: false, RIGHT: true}` to calculate `offsetX`. What do you think about something like `keyEvent.offsetX = direction === "RIGHT" ? elm.width() : 0;`?

Reply:
@cindyli Hi Cindy, that makes sense, and less confusing.